src/lib.rs

```diff
@@ -53,23 +53,32 @@ impl GpsdConnection {
         Ok(Self { inner, raw_data: false })
     }
     /// Enable or disable watcher mode.
-    pub fn watch(&mut self, watch: bool, json: bool, raw: u8) -> GpsdResult<()> {
+    fn _watch(&mut self, watch: bool, json: bool, raw: u8) -> GpsdResult<()> {
         let stream = self.inner.get_mut();
         let watch_data = json!({
             "class": "WATCH",
             "enable": watch,
             "json": json,
             "raw": raw,
         });
-        self.raw_data = if raw > 0 {
-            true
-        } else {
-            false
-        };
+        self.raw_data = raw > 0;
         let msg = format!("?WATCH={}\n", watch_data.to_string());
         stream.write_all(msg.as_bytes())?;
         Ok(())
     }
+    /// Enable or disable watcher mode.
+    pub fn watch(&mut self, watch: bool) -> GpsdResult<()> {
+        self._watch(watch, false, 0)
```

Comment:
Hang on, I think I screwed up my example code - this `false` here should be `true`, right? (because we want JSON for normal watcher mode). Sorry about that; if you could just correct that one nit, I'll merge this.